install_files/securedrop-app-code/debian/postinst

```diff
@@ -97,6 +97,12 @@ case "$1" in
         echo allow-loopback-pinentry > /var/lib/securedrop/keys/gpg-agent.conf
     fi
 
+    # Migrate private keyring to gpg2.1 if needed
+    if [ ! -d "/var/lib/securedrop/keys/private-keys-v1.d" ]; then
```

Comment:
This is a good approach for the upgrade story, in future we can remove these postinst steps (after a few releases)

Reply:
We probably never can. We had instances that were pre 0.4 recently, and that release was a looooooong time ago.

Reply:
So while we have instances that had not ran their Ansible playbooks since before 0.4, the server side was up to latest. Since we can monitor what version the securedrop-app-code package is on via the source interface metadata endpoint (and soon we'll be able to monitor also the base OS via the source interface metadata endpoint), we should be able to remove this from postinst once we have everyone on xenial and post-0.12.0.